symphony-bdk-http/symphony-bdk-http-jersey2/src/main/java/com/symphony/bdk/http/jersey2/ApiClientJersey2RequestLogFilter.java

```diff
@@ -0,0 +1,51 @@
+package com.symphony.bdk.http.jersey2;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.apiguardian.api.API;
+import org.slf4j.Logger;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * This custom request/response filter allows to output the total request time under log 'com.symphony.bdk.requests.outgoing'.
+ * This will only be available is logging level is set to 'DEBUG' in your implementation logger configuration.
+ */
+@Provider
+@API(status = API.Status.INTERNAL)
+public class ApiClientJersey2RequestLogFilter implements ClientRequestFilter, ClientResponseFilter {
+
+  private static final Logger log = getLogger("com.symphony.bdk.requests.outgoing");
+
+  private static final String INTERNAL_CLIENT_START_TIME_REQUEST_HEADER = "X-BDK-internal-request-start-time";
+  private static final String INTERNAL_REQUEST_LOG_MESSAGE = "status={}, url={}, time={}";
+
+  /**
+   * Set the current time in headers when request is sent.
+   */
+  @Override
+  public void filter(ClientRequestContext requestContext) {
+    if (log.isDebugEnabled()) {
+      requestContext.getHeaders().add(INTERNAL_CLIENT_START_TIME_REQUEST_HEADER, System.currentTimeMillis());
```

Comment:
isn't going to end being sent with the request? should we use properties instead? (also part of the request context)

Reply:
Yep, it works with request attributes too :)